contracts/schemes/Competition.sol

```diff
@@ -0,0 +1,349 @@
+        suggestion.votes[msg.sender] = suggestion.votes[msg.sender].add(1);
+        Avatar avatar = ContributionRewardExt(proposal.contributionRewardExt).avatar();
+        uint256 reputation = avatar.nativeReputation().balanceOfAt(msg.sender, proposals[proposalId].snapshotBlock);
+        require(reputation > 0, "voter has no reputation");
+        suggestion.totalVotes = suggestion.totalVotes.add(reputation);
+        suggestion.votes[msg.sender] = reputation;
+        proposal.tiesSuggestions[suggestion.totalVotes] = proposal.tiesSuggestions[suggestion.totalVotes].add(1);
+        refreshTopSuggestions(proposalId, _suggestionId);
+        emit NewVote(proposalId, _suggestionId, msg.sender, reputation);
+        return true;
+    }
+
+    /**
+    * @dev redeem a winning suggestion reward
+    * @param _suggestionId suggestionId
+    * @param _beneficiary - the reward beneficiary.
+    *        this parameter is take into account only if the msg.sender is the suggestion's suggester,
+    *        otherwise the _beneficiary param is ignored and the beneficiary is suggestion's suggester.
+    */
+    function redeem(uint256 _suggestionId, address payable _beneficiary) external {
+        address payable beneficiary = suggestions[_suggestionId].suggester;
+        if (msg.sender == suggestions[_suggestionId].suggester) {
+            //only suggester can redeem to other address
+            if (beneficiary != address(0)) {
+                beneficiary = _beneficiary;
+            }
+        }
+        _redeem(_suggestionId, beneficiary);
+    }
+
+    /**
+    * @dev setSnapshotBlock set the block for the reputaion snapshot
+    * @param _proposalId the proposal id
+    */
+    function setSnapshotBlock(bytes32 _proposalId) public {
+        // solhint-disable-next-line not-rely-on-time
+        require(proposals[_proposalId].votingStartTime < now, "voting period not started yet");
+        if (proposals[_proposalId].snapshotBlock == 0) {
+            proposals[_proposalId].snapshotBlock = block.number;
+            emit SnapshotBlock(_proposalId, block.number);
+        }
+    }
+
+    /**
+    * @dev getOrderedIndexOfSuggestion return the index of specific suggestion in the winners list.
+    * @param _proposalId proposal id
+    * @param _suggestionId suggestion id
+    */
+    function getOrderedIndexOfSuggestion(bytes32 _proposalId, uint256 _suggestionId)
+    public
+    view
+    returns(uint256 index) {
+        uint256[] memory topSuggestionsOrdered = proposals[_proposalId].topSuggestionsOrdered;
+        /** get how many elements are greater than a given element*/
+        for (uint256 i; i < topSuggestionsOrdered.length; i++) {
+            if (suggestions[topSuggestionsOrdered[i]].totalVotes > suggestions[_suggestionId].totalVotes) {
+                index++;
+            }
+        }
+    }
+
+    /**
+    * @dev refreshTopSuggestions this function maintain a winners list array.
```

Comment:
Perhaps be more specific, and write, refreshTopSuggestions will check if the given suggestion is among the top suggestions, and if so, update the  list of top suggestions